content/en/blog/_posts/2023-12-14-disabling-in-tree-cloud-provider-goes-beta.md

```diff
@@ -0,0 +1,210 @@
+taking effect in Kubernetes version 1.29, read on to learn if you are affected
+and what changes you will need to make.
+
+These updated default settings affect a large proportion of Kubernetes users,
+and **will require changes** for users who were previously using the in-tree
+provider integrations. The legacy integrations offered compatibility with
+Azure, AWS, GCE, OpenStack, and vSphere; however for AWS and OpenStack the
+compiled-in integrations were removed in Kubernetes versions 1.26 and 1.27,
+respectively.
+
+## What has changed?
+
+At the most basic level, two [feature gates][fg] are changing their default
+value from false to true. Those feature gates, `DisableCloudProviders` and
+`DisableKubeletCloudCredentialProviders`, control the way that the
+[kube-apiserver][kapi], [kube-controller-manager][kcm], and [kubelet][kubelet]
+invoke the cloud provider related code that is included in those components.
+With these feature gates defaulting to true, the former behavior that was
+provided through the `--cloud-provider` command line flag is changing to only
+recognize the `external` value.
+
+Let's see what the [official Kubernetes documentation][fg] says about these
+feature gates:
+
+> `DisableCloudProviders`: Disables any functionality in `kube-apiserver`,
+> `kube-controller-manager` and `kubelet` related to the `--cloud-provider`
+> component flag.
+
+> `DisableKubeletCloudCredentialProviders`: Disable the in-tree functionality
+> in kubelet to authenticate to a cloud provider container registry for image
+> pull credentials.
+
+## What do you need to do?
+
+If you are upgrading from Kubernetes 1.28+ and are not on Azure, GCE, or
+vSphere then you are in luck, there are no changes you will need to make. If
+you **are** on Azure, GCE, or vSphere, or you are upgrading from a version
+older than 1.28, then read on.
+
+Historically, Kubernetes has included code for a set of cloud providers that
+included AWS, Azure, GCE, OpenStack, and vSphere. Since the inception of
+[KEP-2395][kep2395] the community has been moving towards removal of that
+cloud provider code. The OpenStack provider code was removed in version 1.26,
+and the AWS provider code was removed in version 1.27. This means that users
+who are upgradging from one of the affected cloud providers and versions will
+need to modify their deployments.
+
+### Upgrading on Azure, GCE, or vSphere
+
+There are two options for upgrading in this configuration: migrate to external
+cloud controller managers, or continue using the in-tree provider code.
+Although migrating to external cloud controller managers is recommended,
+there are scenarios where continuing with the current behavior is desired.
+Please choose the best option for your needs.
+
+#### Migrate to external cloud controller managers
+
+Migrating to use external cloud controller managers is the recommended upgrade
+path as it aligns with the future of Kubernetes. To do this you will need to
```

Comment:
```suggestion
path, when possible in your situation. To do this you will need to
```

I doubt the end user cares about "the future of Kubernetes" so perhaps making it more clear when this is relevant to them?

Reply:
I think you are right that they may care about _future changes that will affect them_, but I suspect that "some point some time, undefined, don't worry about it for now" goes directly into the "I do not need to deal with this" category.